File: internal/app/model/place.go
package model

import (
	"time"

	"gorm.io/gorm"
)

type Place struct {
	gorm.Model
	ID      int        `json:"id"`
	UserID  int        `json:"userid"`
	Name    *string    `json:"name"`
	Desc    string     `json:"desc"`
	Cover   string     `json:"cover"`
	Private bool       `json:"private"`
	Group   string     `json:"group" gorm:"default:''"`
	Date    *time.Time `json:"date"`
	Photos  []Photo    `json:"photos"`
	Lon     float64    `json:"lon"`
	Lat     float64    `json:"lat"`
}

File: frontend/src/types/place.ts
export interface Photo {
  id: number
  placeId: number
  url: string
}

export interface Place {
  id: number
  userid: number
  name?: string
  desc: string
  cover: string
  private: boolean
  group: string
  date?: Date
  photos: Photo[]
  lon: number
  lat: number
}

export interface PlaceAddForm {
  name: string
  desc: string
  lon: number
  lat: number
  cover: string
  group: string
  photos: string[]
}

export interface PlaceUpdateForm {
  name: string
  desc: string
  lon: number
  lat: number
  group: string
}
